pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,162 @@
+- **Plugins and plugin configuration.**
+- **Matrix configuration.** The matrix configuration is signed as a whole rather than each individual matrix job. This means the signature is the same for each job in the matrix. When signatures are verified for matrix jobs, the agent double-checks that the job it received is a valid construction of the matrix and that the signature matches the matrix configuration.
+- **The step key.** If there is a step key present. [WIP]
+
+Additionally, the following aspects of the build for each job are included in the signature:
+
+- The pipeline's UUID [WIP]
+- The organization UUID [WIP]
+- The build number [WIP]
+- The repo the build is running for [WIP]
+- The commit the build is running for [WIP]
+
+## Enabling signed pipelines on your agents
+
+You'll need to configure your agents and update pipeline definitions to enable signed pipelines.
+
+### 1. Generate a key pair
+
+Behind the scenes, signed pipelines use [JSON Web Signing (JWS)](https://datatracker.ietf.org/doc/html/rfc7797) to generate signatures. You'll need to generate a [JSON Web Key Set (JWKS)](https://datatracker.ietf.org/doc/html/rfc7517) to sign and verify your pipelines with, then configure your agents to use those keys.
+
+Luckily, the agent has you covered! A JWKS generation tool is built into the agent, which you can use to generate a key pair for you. To use it, you'll need to [install the agent on your machine](/docs/agent/v3/installation), and then run:
+
+```bash
+buildkite-agent tool keygen --alg <algorithm> --key-id <key-id>
+```
+
+Replacing the following:
+
+- `<algorithm>` with the signing algorithm you want to use.
+- `<key-id>` with the key ID you want to use.
+
+For example, to generate an EdDSA key pair with a key ID of `my-key-id`, you'd run:
+
+```bash
+buildkite-agent tool keygen --alg EdDSA --key-id my-key-id
+```
+
+The agent generates two JWKS in your current directory: one private and one public. You can then use these keys to sign and verify your pipelines.
+
+If no key ID is provided, the agent will generate a random one for you.
+
+Note that the value of `--alg` must be a valid [JSON Web Signing Algorithm](https://datatracker.ietf.org/doc/html/rfc7518#section-3). The agent does not support the `none` algorithm, or any signature algorithms in the `RS` series (any signing algorithms that use RSASSA-PKCS1 v1.5 signatures, `RS256` for example).
+
+<details>
+  <summary>Why doesn't the agent support RSASSA-PKCS1 v1.5 signatures?</summary>
+  <p>In short, RSASSA-PKCS1 v1.5 signatures are less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures, which are more secure than RSA signatures.</p>
```

Comment:
I went with the stronger language here, but happy to go back

Reply:
this is fine, i think - the reason i hedged so much is because i wanted to avoid cryptography nerds telling me i was wrong, but i'm _pretty sure_ everything written in there is either objectively correct, or a matter of interpretation